agent/agent.go

```diff
@@ -178,6 +179,10 @@ type Agent struct {
 	// In-memory sink used for collecting metrics
 	MemSink *metrics.InmemSink
 
+	// Eventer provides a backend for handling event logging. APIs are provided on the agent for interacting with
+	// this reloadable type
+	Eventer atomic.Value
```

Comment:
Since this value is only used in enterprise would it be good to create an "EnterpriseAgent" struct and embed it here instead of having the raw value?

That is what we do with the server:

https://github.com/hashicorp/consul/blob/e9e8c0e730ebe546d9aba16d67d7dee92129fb82/agent/consul/server.go#L299-L300

https://github.com/hashicorp/consul/blob/e9e8c0e730ebe546d9aba16d67d7dee92129fb82/agent/consul/enterprise_server_oss.go#L20

That way if we make future additions to the agent that are enterprise specific then we can just add them into that other file.

Reply:
Yeah, that makes sense - I'm working on revising it now.